src/test/java/uk/gov/hmcts/probate/service/notification/AutomatedNotificationPersonalisationServiceTest.java

```diff
@@ -153,4 +154,39 @@ void getPersonalisationShouldIncludeAllFields() {
                         p.get("link_to_case"))
         );
     }
+
+    @Test
+    void getStopReasonShouldhandleNulls() {
+        when(stopReasonService.getStopReasonDescription(LanguagePreference.ENGLISH, "R1"))
+            .thenReturn("Reason One");
+        when(stopReasonService.getStopReasonDescription(LanguagePreference.ENGLISH, "DocumentsRequired"))
+            .thenReturn("Doc XYZ");
+        when(stopReasonService.getStopReasonDescription(LanguagePreference.ENGLISH, "SUB_NULL"))
+            .thenReturn(null); // sub-reason returns null
+
+        StopReason r1 = StopReason.builder()
+            .caseStopReason("R1")
+            .build();
+        StopReason r2 = StopReason.builder()
+            .caseStopReason("DocumentsRequired")
+            .caseStopSubReasonDocRequired("SUB_NULL")
+            .build();
+        StopReason r3 = StopReason.builder()
+            .caseStopReason("DocumentsRequired")
+            .caseStopSubReasonDocRequired(null)
+            .build();
+        CollectionMember<StopReason> cm1 =
+                new CollectionMember<>(null, r1);
+        CollectionMember<StopReason> cm2 =
+                new CollectionMember<>(null, r2);
+        CollectionMember<StopReason> cm3 =
+                new CollectionMember<>(null, r3);
+        List<CollectionMember<StopReason>>
+                domainList = List.of(cm1, cm2, cm3);
```

Comment:
i don't think we need to be quite so aggressive on line breaks here.
also there's a one-arg constructor for `CollectionMember` which implicitly sets the `id` to `null` so maybe that's preferable
```suggestion
        CollectionMember<StopReason> cm1 = new CollectionMember<>(r1);
        CollectionMember<StopReason> cm2 = new CollectionMember<>(r2);
        CollectionMember<StopReason> cm3 = new CollectionMember<>(r3);
        List<CollectionMember<StopReason>> domainList = List.of(cm1, cm2, cm3);
```

Reply:
i'm also wondering if we should verify that if there are non-null values after the null it still includes - we don't currently have anything which gets included but if we just stuck a new value at start and end then that might highlight that if there are multiple stop reasons for the same document they get included multiple times

Reply:
added changes